lib/net/pool.js

```diff
@@ -2012,6 +2130,157 @@ class Pool extends EventEmitter {
     peer.sendCFCheckpt(packet.filterType, packet.stopHash, filterHeaders);
   }
 
+  /**
+   * Handle peer `CFCheckpt` packet.
+   * @method
+   * @private
+   * @param {Peer} peer
+   * @param {CFCheckptPacket} packet
+   */
+
+  async handleCFCheckpt(peer, packet) {
+    if (!this.options.neutrino) {
+      peer.ban();
+      peer.destroy();
+    }
+  }
+
+  /**
+   * Handle peer `CFHeaders` packet.
+   * @method
+   * @private
+   * @param {Peer} peer - Sender.
+   * @param {CFHeadersPacket} packet - Packet to handle.
+   * @returns {void}
+   */
+  async handleCFHeaders(peer, packet) {
+    this.logger.info('Received CFHeaders packet from %s', peer.hostname());
+    if (!this.options.neutrino) {
+      peer.ban();
+      peer.destroy();
+      return;
+    }
+
+    const filterType = packet.filterType;
+
+    if (filterType !== this.requestedFilterType) {
+      this.logger.warning('Received CFHeaders packet with wrong filterType');
+      peer.ban();
+      peer.destroy();
+      return;
+    }
+
+    const stopHash = packet.stopHash;
+    if (!stopHash.equals(this.requestedStopHash)) {
+      this.logger.warning('Received CFHeaders packet with wrong stopHash');
+      peer.ban();
```

Comment:
Let's add a soft ban on the peer. maybe 10.

Reply:
resolved